Application/EdFi.Ods.Api/Configuration/AutoEncryptingOdsInstanceConfigurationDataProviderDecorator.cs

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Data.Common;
+using System.Threading.Tasks;
+using EdFi.Admin.DataAccess.Providers;
+using EdFi.Ods.Api.Providers;
+using EdFi.Ods.Api.Security.Authentication;
+
+namespace EdFi.Ods.Api.Configuration;
+
+/// <summary>
+/// Implements a decorator for <see cref="IEdFiAdminRawOdsInstanceConfigurationDataProvider"/> that automatically encrypts connection strings in the Admin Database.
+/// </summary>
+public class AutoEncryptingOdsInstanceConfigurationDataProviderDecorator : IEdFiAdminRawOdsInstanceConfigurationDataProvider
+{
+    private readonly ISymmetricStringEncryptionProvider _symmetricStringEncryptionProvider;
+    private readonly ISymmetricStringDecryptionProvider _symmetricStringDecryptionProvider;
+    private readonly IEdFiOdsConnectionStringWriter _edFiOdsConnectionStringWriter;
+    private readonly IEdFiAdminRawOdsInstanceConfigurationDataProvider _edFiAdminRawOdsInstanceConfigurationDataProvider;
+    private readonly Lazy<byte[]> _privateKeyBytesLazy;
+
+    public AutoEncryptingOdsInstanceConfigurationDataProviderDecorator(
+        ISymmetricStringEncryptionProvider symmetricStringEncryptionProvider,
+        ISymmetricStringDecryptionProvider symmetricStringDecryptionProvider,
+        IEdFiOdsConnectionStringWriter edFiOdsConnectionStringWriter, 
+        IAdminDatabaseConnectionStringProvider adminDatabaseConnectionStringProvider,
+        DbProviderFactory dbProviderFactory,
+        Lazy<byte[]> privateKeyBytesLazy)
+    {
+        _symmetricStringEncryptionProvider = symmetricStringEncryptionProvider;
+        _symmetricStringDecryptionProvider = symmetricStringDecryptionProvider;
+        _edFiOdsConnectionStringWriter = edFiOdsConnectionStringWriter;
+        _edFiAdminRawOdsInstanceConfigurationDataProvider = new EdFiAdminRawOdsInstanceConfigurationDataProvider(adminDatabaseConnectionStringProvider, dbProviderFactory);
```

Comment:
The problem with instantiating the class here is that you lose all the benefits and features of container initialization. While the dependencies are injected into this class for passthrough, more advanced behaviors that may be added in the future will be unavailable to this instance. Additionally, if the signature of `EdFiAdminRawOdsInstanceConfigurationDataProvider`'s constructor changes, this code will break and require maintenance.

Finally, the decorator pattern indicates that the decorated component is injected into the decorator as an instance of the interface being implemented. Constructing the concrete class in this manner makes it impossible for this decorator to operate with any other future implementation of the interface, creating an undesirable coupling.